install/kubernetes/cilium/values.yaml

```diff
@@ -405,23 +405,34 @@ enableK8sEventHandover: false
 enableXTSocketFallback: true
 
 encryption:
-  # -- Enable transparent network encryption.
-  enabled: false
+  wireguard:
+    # -- Enable transparent network encryption using Wireguard.
+    enabled: false
+
+    # -- Option to set the wireguard IPv4 subnet. Default set to 172.16.43.0/24
+    # wireguardsubnetv4: 172.16.43.0/24
+
+    # -- Option to set the Wireguard IPv6 subnet. Default set to fdc9:281f:04d7:9ee9::1/64
+    # wireguardsubnetv6: fdc9:281f:04d7:9ee9::1/64
+
+  ipsec:
+    # -- Enable transparent network encryption using IPSec.
+    enabled: false
 
-  # -- Name of the key file inside the Kubernetes secret configured via secretName.
-  keyFile: keys
+    # -- Name of the key file inside the Kubernetes secret configured via secretName.
+    keyFile: keys
 
-  # -- Path to mount the secret inside the Cilium pod.
-  mountPath: /etc/ipsec
+    # -- Path to mount the secret inside the Cilium pod.
+    mountPath: /etc/ipsec
 
-  # -- Name of the Kubernetes secret containing the encryption keys.
-  secretName: cilium-ipsec-keys
+    # -- Name of the Kubernetes secret containing the encryption keys.
+    secretName: cilium-ipsec-keys
 
-  # -- Enable encryption for pure node to node traffic.
-  nodeEncryption: false
+    # -- Enable encryption for pure node to node traffic.
+    nodeEncryption: false
 
-  # -- The interface to use for encrypted traffic.
-  # interface: eth0
+    # -- The interface to use for encrypted traffic.
+    # interface: eth0
```

Comment:
Unfortunately this will be a breaking change. @seanmwinn any ideas how we can introduce this change while allowing users to still use the old helm values?

Reply:
Fair enough.

We could also keep the normal `encryption` for IPSec and add an `encrytionWireguard` section.

Reply:
One solution that comes to mind is to just add ``ipsec.type`` as the only new helm value ('ipsec' or 'wireguard') and abstract the rest in a helper.

Reply:
I think @seanmwinn make a lot of sense given that users probably won't want to enable both ipsec and wireguard.

Reply:
Cheers guys, I will do it this way then.